python/tvm/relay/frontend/tensorflow2.py

```diff
@@ -215,10 +232,134 @@ def from_tensorflow(
+            else:
+                for tl_write_name, idx in _tensor_list_write_ops.items():
+                    if node.op.startswith(tl_write_name):
+                        tl_write_nodes.append((node, idx, sub_func_name, root_node))
+                if node.op.startswith("TensorListStack"):
+                    tl_stack_nodes.append(node)
+        elif node.op.startswith("StatelessWhile"):
+            root_node = node.name
+            cond_fn_name, body_fn_name = [
+                parse_attr(node.attr).get(x).name for x in ["cond", "body"]
+            ]
+            for fn_name in [cond_fn_name, body_fn_name]:
+                subfunction = self._gdef_lib[fn_name]
+                sub_func_name = fn_name
+                for sub_node in subfunction.node:
+                    # bypass const node
+                    if sub_node.op == "Const":
+                        continue
+                    self._tf_node_map[sub_node.name] = sub_node
+                    self._analysis_tensor_list_op(
+                        subfunction,
+                        sub_node,
+                        tl_write_nodes,
+                        tl_stack_nodes,
+                        tl_construct_nodes,
+                        sub_func_name=sub_func_name,
+                        root_node=root_node,
+                    )
+
+    def _infer_static_shape_stack_node(self, tl_stack_nodes):
+        for stack_node in tl_stack_nodes:
+            if len(stack_node.input) < 2:
+                # Stack node does not have shape
+                continue
+            input_shape_name = stack_node.input[1].split(":")[0]
+            input_shape_node = self._tf_node_map[input_shape_name]
+            stack = [self._tf_node_map[stack_node.input[0].split(":")[0]]]
+            in_idx = -1
+            while stack:
+                cnode = stack.pop(0)
+                if not cnode.op.startswith("TensorList"):
+                    if in_idx and cnode.op.startswith("StatelessWhile"):
+                        stack.append(self._tf_node_map[cnode.input[in_idx].split(":")[0]])
+                    else:
+                        for iname in cnode.input:
+                            if self._tf_node_map[iname.split(":")[0]].op.startswith(
+                                "StatelessWhile"
+                            ):
+                                # identify input index based on output index
+                                if iname.split(":")[1]:
+                                    in_idx = int(iname.split(":")[1])
+                            stack.append(self._tf_node_map[iname.split(":")[0]])
+                # identify the corresponding constructor node and add shape to _tensor_list_shapes
+                elif cnode.name != stack_node.name:
+                    if is_tensor_list_constuctor(cnode):
+                        shape_attr = parse_attr(input_shape_node.attr)
+                        if "value" not in shape_attr:
```

Comment:
just for curiosity, for the case 'value' not in shape_attr, does that mean it is a input node without shape?

Reply:
yes, then we will treat it as dynamic shape. It will still work as a unit test but in real model, it might cause problem if following by some ops like conv2D